src/many-abci/src/module.rs

```diff
@@ -55,6 +63,45 @@ fn _many_block_from_tendermint_block(block: tendermint::Block) -> Block {
     }
 }
 
+fn _tm_order_from_many_order(order: SortOrder) -> tendermint_rpc::Order {
+    match order {
+        SortOrder::Ascending => tendermint_rpc::Order::Ascending,
+        SortOrder::Descending => tendermint_rpc::Order::Descending,
+        _ => tendermint_rpc::Order::Ascending,
+    }
+}
+
+fn _tm_query_from_many_filter(
+    filter: RangeBlockQuery,
+) -> Result<tendermint_rpc::query::Query, ManyError> {
+    let mut query = tendermint_rpc::query::Query::default();
+    query = match filter {
+        RangeBlockQuery::Height(range) => {
+            query = match range.start_bound() {
+                Bound::Included(x) => query.and_gte("block.height", *x),
+                Bound::Excluded(x) => query.and_gt("block.height", *x),
+                _ => query,
+            };
+            query = match range.end_bound() {
+                Bound::Included(x) => query.and_lte("block.height", *x),
+                Bound::Excluded(x) => query.and_lt("block.height", *x),
+                _ => query,
+            };
+            query
+        }
+        RangeBlockQuery::Time(_range) => return Err(ManyError::unknown("Unimplemented")),
+    };
+
+    // The default query returns an error (TM 0.35)
+    // Return all blocks
+    // TODO: Test on TM 0.34 and report an issue in TM-rs if reproducible
```

Comment:
Is this TODO blocking this PR? If not, add an issue and link it here.

Reply:
Not blocking. I didn't have time to test it on 0.34; thus not opened an issue yet.